genericsStack/src/queue/MyQueue.java

```diff
@@ -0,0 +1,88 @@
+package queue;
+
+import exceptions.MemoryOutOfBounds;
+
+import java.util.EmptyStackException;
+import java.util.List;
+
+public class MyQueue<T> {
+    private T arr[];
+    private Integer size = 0;
+    private Integer capacity = 2 * size + 1;
+
+    public MyQueue() {
+        this.arr = (T[]) new Object[capacity];
+    }
+
+    public Integer getSize() {
+        return size;
+    }
+
+    private T[] copyArrayAndRaiseSize() {
+        capacity = 2 * size;
+        T[] arr1 = (T[]) new Object[capacity];
+        for (int i = 0; i < size; ++i) {
+            arr1[i] = arr[i];
+        }
+        return arr1;
+    }
+
+    public void push(T item) throws StackOverflowError {
```

Comment:
copied from MyStack codes and because of it we have this line, for sure copying is a big part of real programming Nar jan )) but we need be careful.